### DISCUSSION TITLE
Question about sword rotation

### DISCUSSION THREAD

**Discussion by legendiguess** [fishfolk/jumpy#545] [Q&A] [answered]
@zicklag do you have any ideas why sword rotating like this?
![why](https://user-images.githubusercontent.com/40376451/210249343-a70b22ca-0475-4740-b3b6-ab0d6cadcd82.gif)
(i just enabled rotation like i made it with grenade and kick bomb)

**Comment by emilienlavallee** [comment]
Maybe this has to do with the anchor of the sprite?

**Comment by zicklag** [comment, marked as answer]
Even without rotation, I've noticed that for some reason, if I pick up and grab a sword over and over, you can see the flash of a frame where the sword get's parented/deparented to the player, but it doesn't update the sword animation immediately, and you can see the sword either floating above the player's head when picking up ( for just a single frame ) or floating below the players feet when dropping.

I'm not sure if that's related or not.

---

Actually looking at the Gif, I think that the sword is flinging around because the at-rest sword sprite isn't centered in it's bounding box:

![image](https://user-images.githubusercontent.com/25393315/210250139-32bc8a00-494c-49a7-8875-3a8077bf2906.png)

Maybe we need to center the sword in that box?

I think it'd still rotate, so we'd need to figure out how to make that stop rotating when dropped in place, but maybe it wouldn't rotate if the collision box was centered with the sprite?

**Reply by legendiguess** [reply]
You was right, this is a root of a problem, thanks 😄
